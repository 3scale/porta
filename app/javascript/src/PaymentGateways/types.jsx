// @flow

export type BraintreeFormProps = {
  // eslint-disable-next-line flowtype/no-weak-types
  braintreeClient: any,
  billingAddress?: string,
  threeDSecureEnabled: boolean,
  formActionPath: string,
  countriesList: string,
  selectedCountryCode?: string
}
