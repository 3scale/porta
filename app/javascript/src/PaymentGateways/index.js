// @flow

export { StripeFormWrapper } from './stripe/components/StripeFormWrapper'
export { StripeCardForm } from './stripe/components/StripeCardForm'
export { BraintreeForm } from './braintree/BraintreeForm'
export { BraintreeFormWrapper } from './braintree/BraintreeFormWrapper'
export { BraintreeCardFields } from './braintree/BraintreeCardFields'
export { BraintreeBillingAddressFields } from './braintree/BraintreeBillingAddressFields'
export { BraintreeUserFields } from './braintree/BraintreeUserFields'
export { BraintreeSubmitFields } from './braintree/BraintreeSubmitFields'
export * from './braintree/braintree'
export { Label } from './braintree/components/Label'
export { Input } from './braintree/components/Input'
export { ListItem } from './braintree/components/ListItem'
export * from './braintree/types'
