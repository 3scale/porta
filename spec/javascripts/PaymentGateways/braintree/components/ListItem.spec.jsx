import React from 'react'
import { mount } from 'enzyme'
import { ListItem } from 'PaymentGateways'

const Child = () => {
  return <p id="child">Hi</p>
}

const props = {
  id: 'my-unique-id',
  children: null
}

it('should render properly', () => {
  const wrapper = mount(<ListItem {...props} />)
  expect(wrapper.exists()).toBe(true)
})

it('should render without children', () => {
  const wrapper = mount(<ListItem {...props} />)
  expect(wrapper.find('#child').exists()).toBe(false)
})

it('should render children', () => {
  const propsChildren = { id: 'my-unique-id', children: <Child /> }
  const wrapper = mount(<ListItem {...propsChildren} />)
  expect(wrapper.find('#child').exists()).toBe(true)
})
