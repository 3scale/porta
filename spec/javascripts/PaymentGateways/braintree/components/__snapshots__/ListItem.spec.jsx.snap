// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<ListItem
  id="my-unique-id"
>
  <li
    className="string optional form-group"
    id="my-unique-id"
  />
</ListItem>
`;
